src/lib/plot/hint.js

```diff
@@ -199,17 +285,18 @@ class Hint extends PureRenderComponent {
   _getPositionInfo() {
     const {
       value,
-      orientation: initialOrientation} = this.props;
+      horizontalAlign,
+      verticalAlign
+    } = this.props;
 
     const x = getAttributeFunctor(this.props, 'x')(value);
     const y = getAttributeFunctor(this.props, 'y')(value);
 
-    const orientation = initialOrientation === ORIENTATION_AUTO ?
-      this._getOrientationFromAuto(x, y) : initialOrientation;
+    const align = this._getAlign(x, y);
 
     return {
-      style: this._getOrientationStyle(orientation, x, y),
-      className: this._getOrientationClassName(orientation)
+      style: this._getAlignStyle(align, x, y),
```

Comment:
How about optionally taking a prop to replace this function call with something like `this.props.getPositionStyle(this.props, x, y)`.

The default prop for the `getPositionStyle` will be provided by us with these align/orientation capabilities. However, they'll be overridable.